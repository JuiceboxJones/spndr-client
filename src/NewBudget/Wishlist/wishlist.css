.wishlist_page_container{
  padding: 15px;
}
.wishlist_input_container{
width: 50%;
position: relative;
margin: 0 auto;
margin-bottom: 25px;
}
.wishlist_form_container{
  margin: 0 auto;
  width: 25%;
  margin-top: 5%;
  border-radius: 8px 8px 0 0;
  background: #f5f5f5;
  padding: 20px 20px 0 20px;
  height: 50%;
}

#add_wishlist{
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
  margin: 20px;
}
#add_wishlist:hover{
  background: #024e2e9c;
  color:white;
}
.list_container{
  position: relative;
  margin: 0 auto;
  border-radius: 0 0 8px 8px;
  background: #f5f5f5;
  padding: 20px;
  width: 25%;
}

.list_container li{
  border-radius: 2px;
  background: #dbdada;
  font-size: 17px;
  height: 25px;
  font-weight: bold;
  padding: 1px;
  margin-bottom: 10px;
}
.list_container ul{
  margin-top: 0;
}

#home_button{
  border: none;
  width: 150px;
  height: 25px;
  border-radius: 5px;
  background:#A7C4C0;
  cursor: pointer;
  font-weight: bold;
  margin: 20px;
}


#home_button:hover{
  background: #024e2e9c;
  color:white;
}

@media screen and (max-width: 480px) {
  .wishlist_form_container{
    width: 80%;
  }
  .list_container{
    width: 80%;
  }
  #delete{
    margin-top: 0.5px;
  }
  .list_container li{
    text-align: left;
  }
  li a{
    margin-left: 15px;
  }
  #add_wishlist{
    margin: 0;
    margin-top: 15px;
  }
}

@media screen and (max-width: 330px) {
  .wishlist_input_container{
    width: 65%;
  }
  .list_container li{
    font-size: 14px;
    height: 20px;
  }
}