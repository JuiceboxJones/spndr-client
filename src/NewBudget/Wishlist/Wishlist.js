import React, { Component } from 'react';
import ApiService from '../../services/api-fetch-services';
import HistoryHelper from '../../History/History'
import './wishlist.css'

class Wishlist extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {},
    },
  }
  
  state = {
    wishlist: []
  };

  componentDidMount() {
    ApiService.getWishlist().then(data => {
      this.setState({ wishlist: data });
    });
  }

  handleDeleteitem = (e) => {
    const wishlist = this.state.wishlist.filter(item => item.id !== e)
    this.setState({
      wishlist
    })
  }

  handleDelete(e) {
    ApiService.deleteWish(e)
    this.handleDeleteitem(e)
  }

  handleAddToWishlist = e => {
    e.preventDefault(e.item);
    const item = {
      name: e.target.item_name.value,
      url: e.target.item_url.value,
      price: e.target.item_price.value
    };
    ApiService.postWishlist(item).then(item => {
      this.setState({ wishlist: [...this.state.wishlist, item] });});
    e.target.reset();
    e.target.item_name.focus()
  };

  handleNext(){
    HistoryHelper.historyGoTo('/summary', this.props)
  }
  handleBack(){
    HistoryHelper.historyGoBack(this.props)
  }

  handleButtonRender(){
    if(this.props.location.state.prev === `/welcome`){
      return (
        <div className='nextButton'>
          <button type='button' id='home_button' onClick={() => this.handleBack()}>Home</button>
        </div>
      )
    }else{
      return (
        <div className='nextButton'>
          <button type='button' id='back_button' onClick={() => this.handleBack()}>Back</button>
          <button type='button' id='next_button' onClick={() => this.handleNext()}>Next</button>
        </div>
      )
    }
  }

  render() {

    this.handleButtonRender()
    const handleList = this.state.wishlist.map((item, index) => {
      return (
        <li key={item.id || index }>
          <a href={item.url}>{item.name}</a> <span id='wishlist_spacing'/>${item.price} <button type='button' id='delete' onClick={() => this.handleDelete(item.id)}>️
          <span id='delete' role="img" aria-label="delete">❌</span></button>
          <p />
        </li>
      );
    });

    return (
      <div className='wishlist_page_container'>
        <header id='header'>
          <h2>Make a Wishlist</h2>
          <p>Step 3 of 3</p>
        </header>
        <div className='wishlist_form_container'>
        <form className="wishlist_form" onSubmit={e => this.handleAddToWishlist(e)}
        >
          <div className='wishlist_input_container'>
          <label htmlFor=''>Item:</label> 
          <input type="text" required name="item_name" id="item_name" placeholder="e.g. 'New Car'"
          />
          <label htmlFor=''>URL:</label> 
          <input type="text" required name="item_url" id="item_url" placeholder="www.buyItHere.com"
          />
          <label htmlFor=''>Price:</label> 
          <input type="number" step='0.01' required name="item_price" id="item_price" placeholder="100.00"
          />
          <button type="submit" id='add_wishlist'>Add to Wishlist</button>
          </div>
        </form>
        </div>
        <div className='list_container'>
          <ul>{handleList}</ul>
          {this.handleButtonRender()}
        </div>
      </div>
    );
  }
}

export default Wishlist;
