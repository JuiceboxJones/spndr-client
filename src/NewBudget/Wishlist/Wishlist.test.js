import React from 'react';
import ReactDOM from 'react-dom';
import Wishlist from './Wishlist';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  const props = {location:{state:{prev: '/welcome'}}};
  ReactDOM.render(<Wishlist {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const props = {location:{state:{prev: '/welcome'}}};
  const tree = renderer
    .create(<Wishlist {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});