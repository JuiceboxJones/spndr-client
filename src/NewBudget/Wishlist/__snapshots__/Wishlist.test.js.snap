// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wishlist_page_container"
>
  <header
    id="header"
  >
    <h2>
      Make a Wishlist
    </h2>
    <p>
      Step 3 of 3
    </p>
  </header>
  <div
    className="wishlist_form_container"
  >
    <form
      className="wishlist_form"
      onSubmit={[Function]}
    >
      <div
        className="wishlist_input_container"
      >
        <label
          htmlFor=""
        >
          Item:
        </label>
        <input
          id="item_name"
          name="item_name"
          placeholder="e.g. 'New Car'"
          required={true}
          type="text"
        />
        <label
          htmlFor=""
        >
          URL:
        </label>
        <input
          id="item_url"
          name="item_url"
          placeholder="www.buyItHere.com"
          required={true}
          type="text"
        />
        <label
          htmlFor=""
        >
          Price:
        </label>
        <input
          id="item_price"
          name="item_price"
          placeholder="100.00"
          required={true}
          step="0.01"
          type="number"
        />
        <button
          id="add_wishlist"
          type="submit"
        >
          Add to Wishlist
        </button>
      </div>
    </form>
  </div>
  <div
    className="list_container"
  >
    <ul />
    <div
      className="nextButton"
    >
      <button
        id="home_button"
        onClick={[Function]}
        type="button"
      >
        Home
      </button>
    </div>
  </div>
</div>
`;
